src/main/java/com/amazon/opendistroforelasticsearch/knn/index/KNNVectorScriptDocValues.java

```diff
@@ -0,0 +1,69 @@
+/*
+ *   Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.knn.index;
+
+import org.apache.lucene.index.BinaryDocValues;
+import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.index.fielddata.ScriptDocValues;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.util.Objects;
+
+public final class KNNVectorScriptDocValues extends ScriptDocValues<float[]> {
+
+    private final BinaryDocValues binaryDocValues;
+    private boolean docExists;
+
+    public KNNVectorScriptDocValues(BinaryDocValues binaryDocValues) {
+        this.binaryDocValues = binaryDocValues;
+    }
+
+    @Override
+    public void setNextDocId(int docId) throws IOException {
+        if (binaryDocValues.advanceExact(docId)) {
+            docExists = true;
```

Comment:
Is this thread safe? I am wondering if this is the correct place to determine if a value exists for that doc.

Reply:
Will Use instance reference to make it thread safe.

Reply:
I see. Refer to comment below about count